### PR TITLE
fix(formula): handle single quotes in formula and conditional steps [TCTC-8504]

### PR DESCRIPTION
**Pull request by fspot**
Single quotes were not handled properly in ast parser, resulting in formula and conditional steps not working properly.

### CODE REVIEW

File: server/src/weaverbird/pipeline/formula_ast/types.py [hunk 1]
```diff
@@ -33,41 +31,3 @@ class Operation(BaseModel):
...
-    elif isinstance(expr, ColumnName):
-        return f"{column_start_seq}{expr.name}{column_end_seq}"
-    elif isinstance(expr, str):
-        # Unquoting
-        return f"{str_quote_seq}{unquote_string(expr)}{str_quote_seq}"
-    # bool is a subtype of int, but we don't want to have 1 and 0 replaced with 'true' or 'false'
-    elif isinstance(expr, int | float) and not isinstance(expr, bool):
-        return expr
-    elif isinstance(expr, bool):
-        return expr if bools_as_py else ("true" if expr else "false")
```
Comment (lukapeschke) [inline server/src/weaverbird/pipeline/formula_ast/types.py:73]: c'était pour debugger et mettre au propre des formulas ça, on peut éventuellement le garder avec un commentaire ?

**Review by lukapeschke — APPROVED** [review]
:ok_hand:

File: server/tests/backends/fixtures/ifthenelse/quotes.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
           "operator": "gt"
         },
         "newColumn": "COND",
-        "then": "'tintin'",
+        "then": "\"tin'tin\"",
         "else": "some ' stuff with \" annoying 'quotes'"
```
Comment (davinov) [inline server/tests/backends/fixtures/ifthenelse/quotes.json:21]: Shouldn't this else example be invalid? (there is column `some` nor `stuff`, etc...)

**Review by davinov — CHANGES_REQUESTED** [review]
Thanks :+1: 

We also may be a bit laxist with some formulas. Can we prevent invalid ones? (invalid = identifers (tokens without quotes) that doesn't refer to a column)

File: server/src/weaverbird/pipeline/formula_ast/types.py (hunk 1 above)
Comment (fspot) [reply server/src/weaverbird/pipeline/formula_ast/types.py:73]: c'est du code mort

File: server/tests/backends/fixtures/ifthenelse/quotes.json (hunk 2 above)
Comment (PrettyWood) [reply server/tests/backends/fixtures/ifthenelse/quotes.json:21]: 100% aligné. IMO il faut qu'on rende bcp plus robuste notre formula step. Et plus tôt on le fait sur la v3 mieux on se portera

**Pull request merged**